packages/itmat-interface/src/graphql/core/permissionCore.ts

```diff
@@ -81,38 +93,39 @@ export class PermissionCore {
         }
     }
 
-    public async userHasTheNeccessaryDataPermission(operation: string, user: IUser, studyId: string, projectId?: string): Promise<Record<string, unknown> | false> {
+    public async userHasTheNeccessaryDataPermission(operation: string, user: IUserWithoutToken, studyId: string, projectId?: string) {
         if (user === undefined) {
             return false;
         }
         const matchAnyString = '^.*$';
         /* if user is an admin then return true if admin privileges includes needed permissions */
         if (user.type === userTypes.ADMIN) {
             return {
-                matchObj: {
-
-                }, hasVersioned: true, raw: {
+                matchObj: [],
+                hasVersioned: true,
+                raw: {
                     subjectIds: [matchAnyString],
                     visitIds: [matchAnyString],
                     fieldIds: [matchAnyString]
                 },
                 uploaders: [],
-                roles: []
+                roles: [],
+                roleraw: []
             };
         }
 
-        const roles = await db.collections.roles_collection.aggregate([
+        const roles = await db.collections.roles_collection.aggregate<IRole>([
```

Comment:
Hi Florian, could you explain more about this comment? 
The reason I add this is because the default type of an aggregation is Document[] which is not expected to infer its properties; and this aggregation does return IRole directly.